src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php

```diff
@@ -80,15 +96,103 @@ public function getConfig()
     {
         $experimentsMaps = $this->getExperimentsMaps();
         $featuresMap = $this->getFeaturesMap($experimentsMaps[1]);
-
+        $attributes = $this->getConfigAttributes();
+        $audiences = $this->getConfigAudiences();
+        $events = $this->getConfigEvents();
         return new OptimizelyConfig(
             $this->revision,
             $experimentsMaps[0],
             $featuresMap,
-            $this->datafile
+            $this->datafile,
+            $this->environmentKey,
+            $this->sdkKey,
+            $attributes,
+            $audiences,
+            $events
         );
     }
     
+    /**
+     * Generates array of attributes as OptimizelyAttribute.
+     *
+     *
+     * @return array of OptimizelyAttributes.
+     */
+    protected function getConfigAttributes()
+    {
+        $attributArray = [];
+        $attributes = $this->projectConfig->getAttributes();
```

Comment:
i dont think we should be using projectConfig inside these methods directly.

Reply:
other wise we have to pass an argument to these functions so it is same either we pass through argument or call it directly inside function